test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -234,9 +239,109 @@ var _ = ginkgo.Describe("Pod controller", ginkgo.Ordered, ginkgo.ContinueOnFailu
 
 			ginkgo.By("checking that the finalizer is removed when the Pod is deleted early")
 			gomega.Expect(k8sClient.Delete(ctx, createdPod)).Should(gomega.Succeed())
-			gomega.Eventually(func(g gomega.Gomega) bool {
-				return apierrors.IsNotFound(k8sClient.Get(ctx, lookupKey, createdPod))
-			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+			gomega.Eventually(func(g gomega.Gomega) error {
+				return k8sClient.Get(ctx, lookupKey, createdPod)
+			}, util.Timeout, util.Interval).Should(testing.BeNotFoundError())
+		})
+
```

Comment:
Let's add a `ginkgo.When("Using single pod")` to wrap the existing tests and `ginkgo.When("Using pod groups")` for the new tests